### PR TITLE
Add OpenCocktail API

### PR DESCRIPTION
**Pull request by BjoernPetersen**
Missing for now:
- Authorization "scope" definition
  - currently not supported by OpenAPI for non-OAuth
  - could be defined in the descriptions for now
- Image upload and access
- Voting/popularity
- Pagination

Related to #2

### CODE REVIEW

File: OpenCocktail.yaml [hunk 1]
```diff
@@ -0,0 +1,660 @@
...
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Cocktail'
+        404:
+          $ref: "#/components/responses/NotFound"
+    delete:
+      summary: Delete a cocktail
+      operationId: deleteCocktail
+      responses:
```
Comment (FelixGail) [inline OpenCocktail.yaml]: What is the response for not existing ids? (same for other delete calls)

File: OpenCocktail.yaml [hunk 2]
```diff
@@ -0,0 +1,660 @@
...
+        200:
+          description: A list of matching cocktail categories
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/CocktailCategory'
+  /cocktail/category/{id}:
```
Comment (FelixGail) [inline OpenCocktail.yaml]: add option to retrieve all cocktails in a category?

File: OpenCocktail.yaml [hunk 3]
```diff
@@ -0,0 +1,660 @@
...
+        404:
+          $ref: "#/components/responses/NotFound"
+    delete:
+      summary: Delete an ingredient
+      operationId: deleteIngredient
+      responses:
+        204:
+          description: Sucessfully deleted
+
+  /ingredient/category:
```
Comment (FelixGail) [inline OpenCocktail.yaml]: add option to retrieve all ingredients in a category?

File: OpenCocktail.yaml (hunk 2 above)
Comment (BjoernPetersen) [reply OpenCocktail.yaml]: I'd say we use the `GET /cocktail?category=42` operation for that. The query (`q`) obviously shouldn't be required then.

File: OpenCocktail.yaml (hunk 1 above)
Comment (BjoernPetersen) [reply OpenCocktail.yaml]: Yes, most error scenarios, including missing authorization or authentication, are missing

File: OpenCocktail.yaml (hunk 2 above)
Comment (FelixGail) [reply OpenCocktail.yaml]: Is it possible to formulate a call in which either `q` or `category` is required?

Comment (BjoernPetersen) [reply OpenCocktail.yaml]: I think it's not possible to formally specify that, we would have to put that in the parameter/operation description. In this case I don't really see a reason to though, we can just allow to get all cocktails. As soon as we implement pagination that shouldn't be a problem.

**Pull request merged**